src/yz_solrq_worker.erl

```diff
@@ -351,7 +351,7 @@ handle_cast(drain_complete, #state{indexqs = IndexQs} = State) ->
     NewIndexQs = dict:fold(
         fun(Index, #indexq{queue = Queue, queue_len = QueueLen, aux_queue = AuxQueue} = IndexQ, IndexQAccum) ->
             NewIndexQ = IndexQ#indexq{
-                queue = queue:join(Queue, AuxQueue),
+                queue = queue:join(AuxQueue, Queue),
```

Comment:
Actually I think you are right, @lehoff .  My original reasoning above is faulty.

AuxQueue contains the items that have been accumulated while the drain is in progress.  Queue technically should be empty, as this code is called while the worker is in the draining state, but only after all entries in all queues have been successfully drained, unless they have been cancelled due to a time out.  So technically it is possible for Queue to be non-empty, and if so, it contains data that failed to drain, and which is therefore older than the accumulated events in AuxQueue.  Nice catch.